### PR TITLE
font-size: labels & buttons with 11X10 font

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: src/screen.c [hunk 1]
```diff
@@ -42,7 +42,7 @@ void screen_print_debug(const char* message, int duration)
     char print[100];
     snprintf(print, sizeof(print), "%s", message);
     UG_ClearBuffer();
-    UG_FontSelect(&font_font_a_9X9);
+    UG_FontSelect(&font_font_a_11X10);
```
Comment (x1ddos) [inline src/screen.c:45]: What about [screen_init](https://github.com/digitalbitbox/bitbox02-firmware/blob/267b9c240d28cfd682bb2708e1aa178f79411736/src/screen.c#L113)?

Comment (benma) [reply src/screen.c:45]: changed it. note the new commit msg:

> Unclear if this has any real effect in the firmware, as most
components that print text select the font again locally.

 > The bootloader uses the same init, and the strings are affected by
this. For now we force it back to 9x9 in the bootloader, as the
'upgrading' label looks more appealing in the original size.

**Comment by benma** [comment]
> Found a couple more references. What about these?
> 
>     * https://github.com/digitalbitbox/bitbox02-firmware/blob/40ce16f79d54eb295e70acd652f33bdf3aed4b12/src/rust/bitbox02/src/lib.rs#L94-L96
> 
>     * unit test: https://github.com/digitalbitbox/bitbox02-firmware/blob/40ce16f79d54eb295e70acd652f33bdf3aed4b12/test/unit-test/test_ugui.c
> 
>     * all device tests are using the old font but last time I tried only a couple of them actually worked; maybe I'll update it myself separately

fixed (all except screen_print_debug where smaller is better)

File: src/screen.c (hunk 1 above)
Comment (x1ddos) [reply src/screen.c:45]: ah, right. keeping 9X9 for the bootloader makes sense.

File: test/unit-test/test_ugui.c [hunk 2]
```diff
@@ -44,7 +44,7 @@ static void _set_pixel(UG_S16 x, UG_S16 y, UG_COLOR color)
 static void _test_ugui_word_wrap(void** state)
 {
     (void)state; /* unused */
-    UG_Init(&gui, _set_pixel, &font_font_a_9X9, 128, 64);
+    UG_Init(&gui, _set_pixel, &font_font_a_11X10, 128, 64);
```
[the remark this answers is not in the document]
Comment (benma) [reply test/unit-test/test_ugui.c:47]: @NickeZ how do you usually fix this stuff? to me the binary font editing is like magic :joy:

**Comment by x1ddos** [comment]
> > Found a couple more references. What about these?
> > ```
> > * https://github.com/digitalbitbox/bitbox02-firmware/blob/40ce16f79d54eb295e70acd652f33bdf3aed4b12/src/rust/bitbox02/src/lib.rs#L94-L96
> > 
> > * unit test: https://github.com/digitalbitbox/bitbox02-firmware/blob/40ce16f79d54eb295e70acd652f33bdf3aed4b12/test/unit-test/test_ugui.c
> > 
> > * all device tests are using the old font but last time I tried only a couple of them actually worked; maybe I'll update it myself separately
> > ```
> 
> fixed (all except screen_print_debug where smaller is better)

the `ug_font_select` in lib.rs feels like a standard function though, not necessarily used for debug. maybe it could accept a font argument?

File: test/unit-test/test_ugui.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (benma) [reply test/unit-test/test_ugui.c:47]: @x1ddos let's handle it in a separate PR, the cut `W` is already merged anyway.

**Comment by x1ddos** [comment]
Oh and guess what,

    src//bitbox02-rust-c/src/lib.rs:(.text.rust_begin_unwind+0x110):
    undefined reference to `font_font_a_9X9'

:)

**Comment by benma** [comment]
> Oh and guess what,
> 
> ```
> src//bitbox02-rust-c/src/lib.rs:(.text.rust_begin_unwind+0x110):
> undefined reference to `font_font_a_9X9'
> ```
> 
> :)

That is quite a nasty issue. I assume that the 9x9 font is link-time optimized in C as there is no reference to it in the device test C code, or something like that.

File: test/unit-test/test_ugui.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (NickeZ) [reply test/unit-test/test_ugui.c:47]: There are two important things, every row is a character in the array and at the bottom you have all the widths of each character. Sometimes you have to change the width, but most often not.

When you find the right row you start changing things until you understand which bit is which pixel :P

**Comment by benma** [comment]
> > > Found a couple more references. What about these?
> > > ```
> > > * https://github.com/digitalbitbox/bitbox02-firmware/blob/40ce16f79d54eb295e70acd652f33bdf3aed4b12/src/rust/bitbox02/src/lib.rs#L94-L96
> > > 
> > > * unit test: https://github.com/digitalbitbox/bitbox02-firmware/blob/40ce16f79d54eb295e70acd652f33bdf3aed4b12/test/unit-test/test_ugui.c
> > > 
> > > * all device tests are using the old font but last time I tried only a couple of them actually worked; maybe I'll update it myself separately
> > > ```
> > 
> > 
> > fixed (all except screen_print_debug where smaller is better)
> 
> the `ug_font_select` in lib.rs feels like a standard function though, not necessarily used for debug. maybe it could accept a font argument?

fixed without a font arg, as that involves enum boilerplate which I didn't want to spend time doing. can improve it later when components are actually moved to rust.

File: test/unit-test/test_ugui.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (x1ddos) [reply test/unit-test/test_ugui.c:47]: > When you find the right row you start changing things until you understand which bit is which pixel :P

Are you serious? :sweat_smile:

[the remark this answers is not in the document]
Comment (benma) [reply test/unit-test/test_ugui.c:47]: https://github.com/digitalbitbox/bitbox02-firmware/pull/488/

you don't wanna see the script i made to fix it :sweat_smile:

**Pull request merged**